src/Entity/FileEntity.php

```diff
@@ -144,12 +166,12 @@ public function getMimeTypeType() {
   public function postSave(EntityStorageInterface $storage, $update = TRUE) {
     parent::postSave($storage, $update);
     // Save file metadata.
-    if (!empty($this->metadata)) {
+    if (!empty($this->getAllMetadata())) {
```

Comment:
Not sure if we should use the method for internal use, I think it is fine to keep the !empty check here. If you use the method, then you can remove the !empty().